jid/escape.go

```diff
@@ -109,19 +109,18 @@ type unescapeMapping struct {
 	transform.NopResetter
 }
 
-// TODO: Be more specific. Only check if it's the starting character in any
-//       valid escape sequence.
-
+// reference link: https://xmpp.org/extensions/xep-0106.html#unescaping
 func ishex(c byte) bool {
```

Comment:
The purpose of the `ishex` function was previously to see if the *starting* character in any two digit string was a letter that could be a hex character, not to see if the byte is one of the things we would escape. Basically we want to check if the charcater is `'2'` through `'5'` inclusive (ie. a valid start charcater of an escape sequence).